core/worker/src/main/java/com/cognifide/aet/worker/drivers/FirefoxWebDriverFactory.java

```diff
@@ -137,8 +149,10 @@ private void setCommonCapabilities(DesiredCapabilities capabilities, FirefoxProf
     capabilities.setCapability("firefox_binary", new File(path).getAbsolutePath());
   }
 
-  private WebDriver getFirefoxDriver(DesiredCapabilities capabilities) {
-    WebDriver driver = new FirefoxDriver(capabilities);
+  private WebDriver getFirefoxDriver(DesiredCapabilities capabilities)
+      throws MalformedURLException {
+    WebDriver driver = StringUtils.isNotBlank(seleniumGridUrl) ? new RemoteWebDriver(
+        new URL(seleniumGridUrl), capabilities) : new FirefoxDriver(capabilities);
```

Comment:
Maybe let's rename `FirefoxDriver` to `LocalFirefoxDriver` ?

Reply:
FirefoxDriver comes from selenium, we would need to create additional class that extends it.
I'm not sure if it's worth it